### DISCUSSION TITLE
Prospect of Implementing Sapling in Rust?

### DISCUSSION THREAD

**Discussion by wmstack** [GitoxideLabs/gitoxide#1660] [General]
One of Facebook's Interesting Open Source projects is Sapling. Sapling is a fork of Mercurial that is used to power some of Facebook's big projects. I find the UX interesting, and I was thinking what are your thoughts on the difficulty of implementing Sapling in Rust. I am not asking about adding support for sapling to this project, but what would it take to do something similar to Gitoxide for Sapling, and how one would go about this undertaking?

I am also quite liking the license of this project, which unlike Git or Sapling, is licensed under MIT/Apache. The planned WASM support is something that I also find interesting.

**Comment by Byron** [comment]
From what I can tell, Sapling already is predominately written in Rust at 44% at the time of writing. I also know that some parts already use `gitoxide` to implement Git object access. 

Besides that, I don't know enough about Sapling to elaborate, but as long as a particular workflow can be mapped onto Git, it can be implemented with the help of `gitoxide`.
